### PR TITLE
Enhance dev and build documentation

### PR DESCRIPTION
**Pull request by satendrapalsingh**
Enhanced the following pages for documentation:
1. contribute/docs/getting-started
2. contribute/docs/building-chainblocks

### CODE REVIEW

File: docs/docs/contribute/code/getting-started.md [hunk 1]
```diff
@@ -6,21 +6,60 @@ license: CC-BY-SA-4.0
...
 
-## How to start
+To contribute code to any of the above projects you'll need to setup your development environment to work with Git, Github, a code editor, Rust, & C++.
 
-You need to have Git, a GitHub account, and a code editor (we recommend VS Code).
+## Development environment
 
-*If you're not familiar with Git and the GitHub workflow please read the [Installations](https://docs.fragcolor.xyz/docs/01.installations/) and [Making changes](https://docs.fragcolor.xyz/docs/02.making-changes/) sections in the Docs Contribution guide first.*
+### Setup Git & Github
+To install Git on your machine (for Windows, Linux, & Mac) refer to the [Git installation](https://docs.fragcolor.xyz/contribute/docs/installation/#git) section.
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/getting-started.md:20]: Nit pick: usually `git` is *lowercase* (`GitHub` is *PascalCase*).

File: docs/docs/contribute/code/getting-started.md [hunk 2]
```diff
@@ -6,21 +6,60 @@ license: CC-BY-SA-4.0
...
 
-To contribute code to any of these programs you need to build them from sources on your local machine.
+You'll also need an active [Github](https://github.com/) account.
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/getting-started.md:22]: Cloning using the `https` protocol for git doesn't require to have any account. They won't be able to fork and/or contribute code without it though.

**Review by Kryptos-FR — CHANGES_REQUESTED** [review]
Just a nit-pick on `git` name casing in the "getting started" document. I'll review "building chainblocks" later.

File: docs/docs/contribute/code/building-chainblocks.md [hunk 3]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
+
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-22.png)
+
+### Build & run the project
+
+Open your favorite terminal (could be within Visual Studio Code) pointing to this repo folder and run the following sequence of commands.
+
+Create a build directory,
+    
     mkdir build
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/building-chainblocks.md:64]: Although 4 spaces is also detected as a code block,I'd rather we use code fences for consistency:

    ```
    mkdir build
    ```

**Review by Kryptos-FR — CHANGES_REQUESTED** [review]
- Fix use of backticks or code fences
- Prefer tabs and textual output to images

File: docs/docs/contribute/code/getting-started.md (hunk 1 above)
Comment (satendrapalsingh) [reply docs/docs/contribute/code/getting-started.md:20]: No, this needs to be fixed of course!
Have corrected this in both pages.

File: docs/docs/contribute/code/getting-started.md (hunk 2 above)
Comment (satendrapalsingh) [reply docs/docs/contribute/code/getting-started.md:22]: Rewrote this sentence to clarify this.

File: docs/docs/contribute/code/getting-started.md [hunk 4]
```diff
@@ -6,21 +6,60 @@ license: CC-BY-SA-4.0
...
+### Get a code editor
+Although you can use any code editor, we recommend Visual Studio Code (aka VS Code). You can can download and install it from [here](https://code.visualstudio.com/download).
+
+*Download the appropriate installtion file for your platform and follow the on-screen instructions.*
+
+### Install C++
+
+If you're using VS Code you can setup C++ and its compiler (GCC, Clang/LLVM) on your machine, by referring to the official VS Code documentation for the same.
+
+- C++ and GCC (MinGW) on [Windows](https://code.visualstudio.com/docs/cpp/config-mingw)
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/getting-started.md:35]: I figured it was a lot like using well-maintained 3rd party code libraries, lol!

File: docs/docs/contribute/code/building-chainblocks.md [hunk 5]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
 
-Make sure you pull this repository with dependencies `git submodule update --init --recursive`
+And run the following commands to update system packages:
 
-Make sure to use the gnu toolchain for rust with `rustup default stable-x86_64-pc-windows-gnu`
+1. This command will do a full update. You might have to reopen the prompt window
+    ```
+    pacman -Syu --noconfirm
+    ``` 
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-1.png)
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:48]: Sorry, massive copy/paste typo!
Fixed it everywhere.

File: docs/docs/contribute/code/building-chainblocks.md [hunk 6]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
 
-2. Run:
-    ```
-    ./emsdk install latest
-    ./emsdk activate latest
-    source emsdk_env.sh
-    ```
-3. Go back to chainblocks directory and run:
-    ```
+ ![Update User Acc Env Var on Windows](assets/build-cb_git-ignore-file.png)
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:84]: Hmm, I see your point. Removed.

File: docs/docs/contribute/code/building-chainblocks.md [hunk 7]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
-    * Usually `C:\msys64\mingw64\bin`
-3. Open your favorite terminal (could be within visual studio code) pointing to this repo folder
-4. Run those commands:
+2. This command will install dependencies:
+    ```
+    pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain mingw-w64-x86_64-cmake mingw-w64-x86_64-boost mingw-w64-x86_64-ninja mingw-w64-x86_64-clang mingw-w64-x86_64-lld
     ```
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-21.png)
+
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-22.png)
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:56]: Fixed

File: docs/docs/contribute/code/building-chainblocks.md [hunk 8]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
+    ![Update User Acc Env Var on Windows](assets/build-cb_use-gnutoolchain.png)
+
+4. For windows, the path of MinGW bin should be added to your system environment PATH variable (Settings > Edit environment variables for your account > User variables for 'user' > Path). This is needed in order to run MinGW from powershell, vscode, etc. The value of this PATH is usually `C:\msys64\mingw64\bin` .
+
+![Update User Acc Env Var on Windows](assets/build-cb_acc-env-var.png)
+
+### Update system packages
 
-### Instructions
+Open mingw prompt (use `mingw64.exe` from msys2 folder).
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:38]: You're right, I missed that. Corrected it now.

File: docs/docs/contribute/code/building-chainblocks.md [hunk 9]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
-    2. `pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain mingw-w64-x86_64-cmake mingw-w64-x86_64-boost mingw-w64-x86_64-ninja mingw-w64-x86_64-clang mingw-w64-x86_64-lld` - to install dependencies
-2. Add mingw bin to your system environment PATH variable (the windows one, control panel), in otder to run from powershell, vscode, etc
-    * Usually `C:\msys64\mingw64\bin`
-3. Open your favorite terminal (could be within visual studio code) pointing to this repo folder
-4. Run those commands:
+2. This command will install dependencies:
+    ```
+    pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain mingw-w64-x86_64-cmake mingw-w64-x86_64-boost mingw-w64-x86_64-ninja mingw-w64-x86_64-clang mingw-w64-x86_64-lld
     ```
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-21.png)
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:54]: Added 'command' and 'output' tabs (with terminal text) and removed the images. No image alt-text anymore.

[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:54]: Also, implementing this across the doc for terminal command/ output scenarios

File: docs/docs/contribute/code/building-chainblocks.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:56]: Images replaced with command/ output tabs (no alt-text anymore).

File: docs/docs/contribute/code/building-chainblocks.md [hunk 10]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
+
+2. Make sure you pull this project repository with dependencies by using the following command (from any terminal): 
+    
+    `git submodule update --init --recursive`
+
+3. Ensure you're using the Rust GNU toolchain by firing the following rustup command (from any terminal):
+
+    `rustup default stable-x86_64-pc-windows-gnu`
+
+    ![Update User Acc Env Var on Windows](assets/build-cb_use-gnutoolchain.png)
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:30]: Image replaced with command/ output tabs (no alt-text anymore).

File: docs/docs/contribute/code/building-chainblocks.md [hunk 11]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
+and then the following command to actually build and link Chainblocks.
+
     ninja format; ninja cbl
-    ```
 
-#### WASM build
+After the completion of the build you'll see a message on the terminal confirming the linking of cbl.exe.
 
-Use a `mingw64.exe` terminal for this.
+ ![Update User Acc Env Var on Windows](assets/build-cb_use-build-link.png)
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:80]: Fixed

File: docs/docs/contribute/code/building-chainblocks.md [hunk 12]
```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
...
+
+1. The following tools need to be available on your system
+    - msys2 - https://www.msys2.org/
+    - rust - gnu toolchain
+
+    *You probably already have these if you've followed the [Getting Started](https://docs.fragcolor.xyz/contribute/code/getting-started/) section - msys2 is installed as part of C++/ MinGW, and rustup installs the required rust toolchain.*
+
+2. Make sure you pull this project repository with dependencies by using the following command (from any terminal): 
+    
+    `git submodule update --init --recursive`
```
[the remark this answers is not in the document]
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:24]: Fixed. Also, fixed everywhere on both pages (as applicable).

File: docs/docs/contribute/docs/installation.md [hunk 13]
```diff
@@ -43,54 +43,72 @@ And choose to disable the Path Length Limit (just in case your default installat
 
 Retain the other default installations settings.
 
-To confirm Python was installed successfully type either Python or Py in the windows command prompt:
+To confirm Python was installed successfully type either Python or Py in the windows command prompt. If Python is Installed it will show the version Details.
 
-> Python
+=== "Command"
+    ```
```
Comment (Kryptos-FR) [inline docs/docs/contribute/docs/installation.md:49]: It's good to leave an empty line before and after the code block. It doesn't change anything during generation but it makes the original file easier to read.

Comment (Kryptos-FR) [reply docs/docs/contribute/docs/installation.md:49]: === "Command"

        ```
        Python
        ```

    === "Result"

File: docs/docs/contribute/code/building-chainblocks.md [hunk 14]
```diff
@@ -5,122 +5,187 @@ license: CC-BY-SA-4.0
...
-    
-    cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=$EMSDK/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake ..
-    
-    ninja format; ninja rust_blocks; ninja cbl
-    
+From the chainblocks directory, run the following commands in sequence from the VS Code terminal to create and link the WASM build (similar to how we built Chainblocks for plain C++, in the previous section).
+```    
+mkdir build-wasm
+```     
+```
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/building-chainblocks.md:179]: Add an empty line between blocks

File: docs/docs/contribute/code/building-chainblocks.md [hunk 15]
```diff
@@ -5,122 +5,187 @@ license: CC-BY-SA-4.0
...
-    mkdir build-wasm
-    
-    cd build-wasm
-    
-    cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=$EMSDK/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake ..
-    
-    ninja format; ninja rust_blocks; ninja cbl
-    
+From the chainblocks directory, run the following commands in sequence from the VS Code terminal to create and link the WASM build (similar to how we built Chainblocks for plain C++, in the previous section).
+```
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/building-chainblocks.md:176]: remove trailing whitespaces

File: docs/docs/contribute/code/building-chainblocks.md [hunk 16]
```diff
@@ -5,122 +5,187 @@ license: CC-BY-SA-4.0
...
-    cd build-wasm
-    
-    cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=$EMSDK/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake ..
-    
-    ninja format; ninja rust_blocks; ninja cbl
-    
+From the chainblocks directory, run the following commands in sequence from the VS Code terminal to create and link the WASM build (similar to how we built Chainblocks for plain C++, in the previous section).
+```    
+mkdir build-wasm
+```
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/building-chainblocks.md:178]: remove trailing whitespaces

File: docs/docs/contribute/code/building-chainblocks.md [hunk 17]
```diff
@@ -5,122 +5,187 @@ license: CC-BY-SA-4.0
...
-    emsdk activate latest
+Activate the latest SDK for the current user.    
+```
+emsdk activate latest
+```
 
 Activate the PATH/ environment variables for the current terminal session.
-    
-    emsdk_env.bat
+```
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/building-chainblocks.md:169]: remove trailing whitespaces

File: docs/docs/contribute/code/building-chainblocks.md [hunk 18]
```diff
@@ -5,122 +5,187 @@ license: CC-BY-SA-4.0
...
 
 Use a `mingw64.exe` terminal to navigate to the emsdk directory,
-    
-    cd emsdk
-
-and then Git Pull to get the latest tools from Github (not required the first time you clone the git).
-
-    git pull
+```
```
Comment (Kryptos-FR) [inline docs/docs/contribute/code/building-chainblocks.md:149]: remove trailing whitespaces

**Review by Kryptos-FR — CHANGES_REQUESTED** [review]
Looks better.

Another nit-pick: add an empty line before/between code blocks to make them easier to maintain.

File: docs/docs/contribute/docs/installation.md (hunk 13 above)
Comment (satendrapalsingh) [reply docs/docs/contribute/docs/installation.md:49]: Agreed. Fixed it everywhere applicable.

File: docs/docs/contribute/code/building-chainblocks.md (hunk 14 above)
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:179]: Fixed everywhere applicable.

File: docs/docs/contribute/code/building-chainblocks.md (hunk 15 above)
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:176]: Fixed across pages wherever applicable.

File: docs/docs/contribute/code/building-chainblocks.md (hunk 16 above)
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:178]: Fixed

File: docs/docs/contribute/code/building-chainblocks.md (hunk 18 above)
Comment (satendrapalsingh) [reply docs/docs/contribute/code/building-chainblocks.md:149]: Fixed

**Review by Kryptos-FR — APPROVED** [review]

**Pull request merged**